src/Glpi/Search/Input/QueryBuilder.php

```diff
@@ -845,9 +847,60 @@ public static function cleanParams(array $params): array
             }
         }
 
+        self::validateCriteria($params);
+
         return $params;
     }
 
+    private static function validateCriteria(array &$params): void
+    {
+        if (!isset($params['criteria'])) {
+            return;
+        }
+        $valid_main_opts = SearchOption::getOptionsForItemtype($params['itemtype']);
+
+        // Validate criteria
+        $invalid_criteria_count = 0;
+        foreach ($params['criteria'] as $k => $criterion) {
+            if (!array_key_exists('field', $criterion) || !is_numeric($criterion['field'])) {
+                continue;
+            }
+            if (isset($criterion['itemtype'])) {
+                // In the criteria array, the search options are from the metatype POV (Agent Name for example is ID 1 in criteria array, but 900 from the POV of Computer)
+                $valid_meta_opts = SearchOption::getOptionsForItemtype($criterion['itemtype']);
+                if (!isset($valid_meta_opts[(int) $criterion['field']])) {
+                    $invalid_criteria_count++;
+                    unset($params['criteria'][$k]);
+                }
+            } else if (!isset($valid_main_opts[(int) $criterion['field']])) {
+                $invalid_criteria_count++;
+                unset($params['criteria'][$k]);
+            }
+        }
+
+        // Validate sorts
+        if (isset($params['sort'])) {
+            if (!is_array($params['sort'])) {
+                $params['sort'] = [(int) $params['sort']];
+                $params['order'] = [$params['order'] ?? 'ASC'];
+            }
+            foreach ($params['sort'] as $k => $sorted_id) {
+                // Validate sort (IDs are always from the POV of the main itemtype)
+                if (!isset($valid_main_opts[$sorted_id])) {
+                    unset($params['sort'][$k], $params['order'][$k]);
+                }
+            }
+        }
+        if (empty($params['sort'])) {
+            $params['sort'] = [0];
+            $params['order'] = ['ASC'];
+        }
+
+        if ($invalid_criteria_count > 0) {
+            // There is probably no need to show more information about the invalid criteria
```

Comment:
Well, for user, probably not; but from a dev point of view; being able to find which option is wrong may be of interest. Maybe just add a log entry when in debug mode or dev environment?

Reply:
Same idea, from user pov it's ok, from dev pov, we need to have log in php-errors for each invalid search options.

Reply:
I don't known, you usually don't want bad request to fill your logs (at least in production).
So OK but for dev env only.